python/src/aiconfig/default_parsers/openai_vision.py

```diff
@@ -0,0 +1,427 @@
+                        add_prompt_as_message(
+                            previous_prompt,
+                            aiconfig,
+                            completion_params["messages"],
+                            params,
+                        )
+        else:
+            # If messages are already specified in the model settings, then just resolve each message with the given parameters and append the latest message
+            for i in range(len(completion_params.get("messages"))):
+                prompt_content = completion_params["messages"][i]["content"]
+                if not prompt_content:
+                    continue
+                if isinstance(prompt_content, str):
+                    completion_params["messages"][i]["content"] = (
+                        resolve_prompt_string(
+                            prompt,
+                            params,
+                            aiconfig,
+                            prompt_content,
+                        )
+                    )
+                else:
+                    # If the prompt content is an array, then it's a ChatCompletionContentPartParam object
+                    for j in range(len(prompt_content)):
+                        if prompt_content[j].get("text"):
+                            prompt_content[j]["text"] = resolve_prompt_string(
+                                prompt,
+                                params,
+                                aiconfig,
+                                prompt_content[j]["text"],
+                            )
+
+        # Add in the latest prompt
+        add_prompt_as_message(
+            prompt, aiconfig, completion_params["messages"], params
+        )
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_deserialize_complete",
+                __name__,
+                {"output": completion_params},
+            )
+        )
+        return completion_params
+
+
+def refine_chat_completion_params(model_settings, aiconfig, prompt):
+    # completion parameters to be used for openai's chat w/ vision completion api
+    # system prompt handled separately
+    # streaming handled separately.
+    # Messages array built dynamically and handled separately
+    supported_keys = {
+        "frequency_penalty",
+        "logit_bias",
+        "max_tokens",
+        "model",
+        "messages",
+        "n",
+        "presence_penalty",
+        "stop",
+        "stream",
+        "temperature",
+        "top_p",
+        "user",
+    }
+
+    completion_data = {}
+    for key in supported_keys:
+        if key in model_settings:
+            completion_data[key] = model_settings[key]
+
+    # Explicitly set the model to use if not already specified
+    if completion_data.get("model") is None:
+        model_name = aiconfig.get_model_name(prompt)
+        completion_data["model"] = model_name
```

Comment:
cc @Ankush-lastmile please I can't take it anymore!!!! Think about the children!